GiniVision/Classes/Core/Screens/Camera/CameraPreviewViewController.swift

```diff
@@ -0,0 +1,292 @@
+    public override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        camera?.stop()
+    }
+    
+    public override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+        
+        coordinator.animate(alongsideTransition: { [weak self] _ in
+            guard let `self` = self else {
+                return
+            }
+            
+            self.updatePreviewViewOrientation()
+        })
+    }
+    
+    fileprivate func updatePreviewViewOrientation() {
+        let orientation: AVCaptureVideoOrientation
+        if UIDevice.current.isIpad {
+            orientation =  interfaceOrientationsMapping[UIApplication.shared.statusBarOrientation] ?? .portrait
+        } else {
+            orientation = .portrait
+        }
+        
+        let previewLayer = (self.previewView.layer as? AVCaptureVideoPreviewLayer)
+        previewLayer?.connection?.videoOrientation = orientation
+    }
+    
+    fileprivate func setupCamera(giniConfiguration: GiniConfiguration) {
+        camera = Camera(giniConfiguration: giniConfiguration) { error in
+            if let error = error {
+                switch error {
+                case .notAuthorizedToUseDevice:
+                    addNotAuthorizedView()
+                default:
+                    if GiniConfiguration.DEBUG {
+                        cameraState = .valid
+                        #if targetEnvironment(simulator)
+                        addDefaultImage()
+                        #endif
+                    }
+                }
+            }
+        }
+                
+        if giniConfiguration.qrCodeScanningEnabled {
+            camera?.setupQRScanningOutput()
+            camera?.didDetectQR = { [weak self] qrDocument in
+                guard let `self` = self else { return }
+                self.delegate?.cameraPreview(self, didDetect: qrDocument)
+            }
+        }
+        
+    }
+    
+    func captureImage(completion: @escaping (Data?, CameraError?) -> Void) {
+        guard let camera = camera else {
+            Log(message: "No camera initialized", event: .warning)
+            return
+        }
+        
+        if GiniConfiguration.DEBUG {
+            // Retrieves the image from default image view to make sure the image
+            // was set and therefore the correct states were checked before.
+            #if targetEnvironment(simulator)
+            if let image = self.defaultImageView?.image,
+                let imageData = image.jpegData(compressionQuality: 0.2) {
+                completion(imageData, nil)
+            }
+            #endif
+        }
+        
+        camera.captureStillImage(completion: { data, error in
+            if let data = data,
+                let image = UIImage(data: data),
+                let imageData = image.jpegData(compressionQuality: 0.9) {
```

Comment:
Why 0.9? Wasn't the optimal compression level 0.7? Or maybe something changed.

Reply:
it was always 0.9, but I would make it 1.0 since it doesn't affect the image size that much